### PR TITLE
Fix SLAM (Antimatter Generator) produce 0 in wireless mode. Add avg efficiency of 10 last cycles.

### PR DESCRIPTION
**Pull request by lordIcocain**
getExoticEnergyHatches() return nothing in checkMachine(), so i moved capacity check to createEU().

Edit:
image
image

### CODE REVIEW

File: src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java [hunk 1]
```diff
@@ -442,8 +459,31 @@ private float getEfficiency() {
         return this.annihilationEfficiency;
     }
 
+    private int n = 0;
+
+    private void setAvgEff(float a) {
+        if (n == 10) n = 0;
+        if (this.avgEff.size() < 10) {
```
Comment (S4mpsa) [inline src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:466]: Isn't `avgEff` constructed with size = 10? Seems redundant to check

Comment (lordIcocain) [reply src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:466]: Its a list, at initial size() will return 0.

File: src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java [hunk 2]
```diff
@@ -442,8 +459,31 @@ private float getEfficiency() {
...
+        } else {
+            this.avgEff.set(n, a);
+            n++;
+        }
+
+        float b = 0;
+        for (float c : this.avgEff) {
+            b += c;
+        }
+        this.avgEffCache = b == 0 ? 0 : b / this.avgEff.size();
```
[the remark this answers is not in the document]
Comment (lordIcocain) [reply src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java:477]: size() will return 0 until you add something in list. then 1, etc. initialSize probably using only for allocate memory. list can go even past initial value.

File: src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java [hunk 3]
```diff
@@ -178,6 +178,12 @@ public CheckRecipeResult checkProcessing() {
         if (i == 2 && containedAntimatter > 0 && catalystFluid != null) {
             createEU(containedAntimatter, catalystFluid);
         }
+        // Set stats if one fluid supplied.
+        if ((containedAntimatter == 0 & catalystFluid != null) | (containedAntimatter > 0 & catalystFluid == null)) {
```
[the remark this answers is not in the document]
Comment (lordIcocain) [reply src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java]: done

**Review by S4mpsa — APPROVED** [review]

**Pull request merged**